### DISCUSSION TITLE
Serverless Keystone with admin UI

### DISCUSSION THREAD

**Discussion by ghost** [keystonejs/keystone#8526] [Questions]
I have been tinkering on how to serve the admin UI on Vercel with the embedded Next.js approach. I followed Josh Calder (@borisno2) video and repository for this:

https://www.youtube.com/watch?v=Wq6eQVyW8ug
https://github.com/borisno2/on-the-hill-drama-club

So far, no luck. I am getting a 404 error on Vercel when I try to access the admin UI. 

My goal is to set up a simple Keystone + Next.js website with internationalization (using Next 13 middleware) that is editable online at no costs. I am currently using Planetscale for the database (free tier) and a AWS s3 bucket for the assets (free tier).

You can replicate my projet here: https://github.com/johnboylesingfield/john-boyle-singfield
Frontend: https://johnboylesingfield.com/
Keystone: https://admin.johnboylesingfield.com/

PS: I know I am early on the serverless option and I understand this is not officially supported yet. I would however love to get any feedback and other possible solutions on the matter. Since I try to set up websites for low income individuals/non-profit in the arts, a system like this would be extremely beneficial.

As always, thanks for the amazing work and help.

**Comment by borisno2** [comment]
Hi @johnboylesingfield

It looks like you are trying to run Keystone Admin UI on Vercel as a completely separate app. Currently, the Admin UI depends on the GraphQL API to run, and the default keystone output does not support running the GraphQL API in Vercel. You will need to set up GraphQL API in your Frontend App and use [NextJS Zones](https://nextjs.org/docs/advanced-features/multi-zones) to have Keystone Admin accessible via https://johnboylesingfield.com/admin (or something similar). 

My guess is you are getting a 404 in your current Keystone Vercel deployment because Vercel is not able to deploy it, and the app is in an Error state in Vercel.

**Reply by ghost** [reply]
Thanks a lot for the reply! I tried to follow your repo more closely to implement NextJS Zones with rewrites. After one day of trying it out, I unfortunately still get a 404 error on GET requests for admin signup.

I tried to disable admin signup completely. When I do that, a Keystone page shows up with the message "Something went wrong / Reload Page". The error in the console is `Error: An error occurred when loading Admin Metadata`.

I also tried removing internationalization completely to see if it was the cause. It does not seem to be.

The error is apparent in Vercel as well as locally.

https://github.com/johnboylesingfield/john-boyle-singfield
https://john-boyle-singfield-keystone.vercel.app/
https://johnboylesingfield.com/

Anyway, I should probably just wait until an official way of doing this is adopted! Sorry, I get too excited with this sort of things :)

Again, thank you for the examples and super helpful videos. Have a great week.
